lms/djangoapps/courseware/access_utils.py

```diff
@@ -103,3 +112,57 @@ def check_course_open_for_learner(user, course):
     if COURSE_PRE_START_ACCESS_FLAG.is_enabled():
         return ACCESS_GRANTED
     return check_start_date(user, course.days_early_for_beta, course.start, course.id)
+
+
+def check_enrollment(user, course):
+    """
+    Check if the course requires a learner to be enrolled for access.
+
+    Returns:
+        AccessResponse: Either ACCESS_GRANTED or EnrollmentRequiredAccessError.
+    """
+    if check_public_access(course, [COURSE_VISIBILITY_PUBLIC]):
+        return ACCESS_GRANTED
+
+    if CourseEnrollment.is_enrolled(user, course.id):
+        return ACCESS_GRANTED
+
+    return EnrollmentRequiredAccessError()
+
+
+def check_authentication(user, course):
+    """
+    Grants access if the user is authenticated, or if the course allows public access.
+
+    Returns:
+        AccessResponse: Either ACCESS_GRANTED or AuthenticationRequiredAccessError
+    """
+    if user.is_authenticated:
+        return ACCESS_GRANTED
+
+    if check_public_access(course, [COURSE_VISIBILITY_PUBLIC]):
+        return ACCESS_GRANTED
```

Comment:
If a course is visible to the public, shouldn't it come before the `is_authenticated` check?